Quazi,

The FC=ifort can be argument given when compiling.  gfortran should be left
as the default, and that command is just an override.

On Thu, Jun 8, 2023, 10:57 PM quaz115 ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/Makefile
> <https://github.com/noaa-oar-arl/canopy-app/pull/75#discussion_r1223782608>
> :
>
> >
>  # Compiler
>  FC ?= gfortran
>  ifeq ($(FC), f77)  # override possible Make default
> +  FC := ifort
>
> @zmoon <https://github.com/zmoon> @drnimbusrain
> <https://github.com/drnimbusrain> , i think we need this change, When i
> revert back, FC shows as gfortran and i get build error:
> Fatal Error: File 'netcdf.mod' opened at (1) is not a GNU Fortran module
> file
> make: *** [canopy_ncf_io_mod.o] Error 1
> I think we need these changes (or FC to be assigned ifort somewhere)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noaa-oar-arl/canopy-app/pull/75#discussion_r1223782608>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGLFYNWE4MLNESZHLHLL53TXKKGKNANCNFSM6AAAAAAY77ZZUM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

--- a/src/Makefile
+++ b/src/Makefile
@@ -6,7 +6,7 @@
 # Compiler
 FC ?= gfortran
 ifeq ($(FC), f77)  # override possible Make default
-  FC := ifort
+  FC := gfortran
 endif
 $(info FC setting: '$(FC)')
 
@@ -22,9 +22,9 @@ ifeq ($(DEBUG), 1)
   ifeq ($(FC),gfortran)
     FCFLAGS := -g -Wall -Wextra -Wconversion -Og -pedantic -fcheck=bounds -fmax-errors=5 
   else ifeq ($(FC),ifort)
-    # Both the following Debug flag options build Canopy-app on Hera with ifort without warnings 
-    #FCFLAGS := -g -warn all -implicitnone -O1 -stand f90 -error-limit 5  #Equivalent to gfortran Debug flags
-    FCFLAGS  := -g -warn all -check bounds -traceback 
+    # Tested the following Debug flag options on Hera with 'DEBUG=1 NETCDF=1 FC=ifort make -C src'
+    #FCFLAGS := -g -warn all -implicitnone -O1 -stand f90 -error-limit 5  #Equivalent to gfortran Debug flags  #builds but with WARNINGS to be resolved 
+    FCFLAGS  := -g -warn all -check bounds -traceback  #builds Canopy-app on Hera with FC=ifort without warnings
   endif
 else ifeq ($(DEBUG), 0)
   FCFLAGS := -O3

--- a/src/Makefile
+++ b/src/Makefile
@@ -36,7 +36,7 @@ endif
 # NETCDF Settings here
 LIBS :=
 INC :=
-$(info NETCDF setting: '$(NC)')
+$(info NC setting: '$(NC)')
 ifeq ($(NC), 1)
   NETCDF_FLIBS := $(shell nf-config --flibs)
   NETCDF_INC := -I$(shell nf-config --includedir)

--- a/src/Makefile
+++ b/src/Makefile
@@ -1,7 +1,7 @@
 #
 # Use `FC=<compiler> ... make` to select compiler
 # Use `DEBUG=1 ... make` for a debug build
-# Use `NETCDF=1 ... make` to build with NetCDF using `nf-config`
+# Use `NC=1 ... make` to build with NetCDF using `nf-config`
 
 # Compiler
 FC ?= gfortran
@@ -14,17 +14,18 @@ $(info FC setting: '$(FC)')
 DEBUG ?= 0
 
 # Default to NetCDF build
-NETCDF ?= 1
+NC ?= 1
 
 # Compile flags
 $(info DEBUG setting: '$(DEBUG)')
 ifeq ($(DEBUG), 1)
   ifeq ($(FC),gfortran)
-    FCFLAGS := -g -Wall -Wextra -Wconversion -Og -pedantic -fcheck=bounds -fmax-errors=5 
+    FCFLAGS := -g -Wall -Wextra -Wconversion -Og -pedantic -fcheck=bounds -fmax-errors=5 -std=f2003 -fall-intrinsics
   else ifeq ($(FC),ifort)
-    # Tested the following Debug flag options on Hera with 'DEBUG=1 NETCDF=1 FC=ifort make -C src'
-    #FCFLAGS := -g -warn all -implicitnone -O1 -stand f90 -error-limit 5  #Equivalent to gfortran Debug flags  #builds but with WARNINGS to be resolved 
-    FCFLAGS  := -g -warn all -check bounds -traceback  #builds Canopy-app on Hera with FC=ifort without warnings
+    # Tested the following Debug flag options on Hera with 'DEBUG=1 NC=1 FC=ifort make -C src'
+    # Both options build without warnings and also make clean in src
+    FCFLAGS := -g -warn all -implicitnone -O1 -error-limit 5  #Equivalent to gfortran Debug flags
+    #FCFLAGS  := -g -warn all -check bounds -traceback  #also builds Canopy-app on Hera with FC=ifort without warnings
   endif
 else ifeq ($(DEBUG), 0)
   FCFLAGS := -O3
@@ -35,18 +36,18 @@ endif
 # NETCDF Settings here
 LIBS :=
 INC :=
-$(info NETCDF setting: '$(NETCDF)')
-ifeq ($(NETCDF), 1)
+$(info NETCDF setting: '$(NC)')
+ifeq ($(NC), 1)
   NETCDF_FLIBS := $(shell nf-config --flibs)
   NETCDF_INC := -I$(shell nf-config --includedir)
   #
   LIBS += $(NETCDF_FLIBS)
   INC += $(NETCDF_INC)
   FCFLAGS += -DNETCDF
-else ifeq ($(NETCDF), 0)
+else ifeq ($(NC), 0)
   #
 else
-  $(error invalid setting for NETCDF, should be 0 or 1 but is '$(NETCDF)')
+  $(error invalid setting for NETCDF, should be 0 or 1 but is '$(NC)')
 endif
 $(info LIBS: '$(LIBS)')
 $(info INC:  '$(INC)')
@@ -80,7 +81,7 @@ OBJS :=\
  canopy_dealloc.o \
  canopy_app.o
 
-ifeq ($(NETCDF), 0)
+ifeq ($(NC), 0)
   _ncf_objs := canopy_check_input.o canopy_ncf_io_mod.o
   OBJS := $(filter-out $(_ncf_objs),$(OBJS))
 endif

